src/lib/cache/SubscriptionCache.cpp

```diff
@@ -0,0 +1,1000 @@
+*
+* subCacheMutexTime - 
+*/
+static struct timespec subCacheMutexTime = { 0, 0 };
+
+
+
+/* ****************************************************************************
+*
+* subscriptionCacheInit - 
+*/
+void subscriptionCacheInit(std::string dbName)
+{
+  subCache = new SubscriptionCache(dbName);
+  subCache->init();
+}
+
+
+
+/* ****************************************************************************
+*
+* subCacheMutexWaitingTimeGet - 
+*/
+void subCacheMutexWaitingTimeGet(char* buf, int bufLen)
+{
+  if (semTimeStatistics)
+  {
+    snprintf(buf, bufLen, "%lu.%09d", subCacheMutexTime.tv_sec, (int) subCacheMutexTime.tv_nsec);
+  }
+  else
+  {
+    snprintf(buf, bufLen, "Disabled");
+  }
+}
+
+
+
+/* ****************************************************************************
+*
+* EntityInfo::EntityInfo - 
+*/
+EntityInfo::EntityInfo(const std::string& idPattern, const std::string& _entityType)
+{
+  regcomp(&entityIdPattern, idPattern.c_str(), 0);
+  entityType = _entityType;
+}
+
+
+
+/* ****************************************************************************
+*
+* match - 
+*/
+bool EntityInfo::match
+(
+  const std::string&  id,
+  const std::string&  type
+)
+{
+  //
+  // If type non-empty - perfect match is mandatory
+  // If type is empty - always matches
+  //
+  if ((type != "") && (entityType != type))
+  {
+    return false;
+  }
+
+  // REGEX-comparison this->entityIdPattern VS id
+  return regexec(&entityIdPattern, id.c_str(), 0, NULL, 0) == 0;
+}
+
+
+
+/* ****************************************************************************
+*
+* Subscription::Subscription - 
+*/
+Subscription::Subscription()
+{
+  tenant                = "";
+  servicePath           = "";
+  subscriptionId        = "";
+  throttling            = 0;
+  expirationTime        = 0;
+  lastNotificationTime  = -1;
+  pendingNotifications  = 0;
+  format                = XML;
+}
+
+
+
+/* ****************************************************************************
+*
+* Subscription::Subscription - 
+*/
+Subscription::Subscription
+(
+  const std::string& _tenant,
+  const std::string& _servicePath,
+  const std::string& _subscriptionId,
+  Format             _format
+)
+{
+  tenant                = _tenant;
+  servicePath           = _servicePath;
+  subscriptionId        = _subscriptionId;
+  throttling            = 0;
+  expirationTime        = 0;
+  lastNotificationTime  = -1;
+  pendingNotifications  = 0;
+  format                = _format;
+}
+
+
+
+/* ****************************************************************************
+*
+* Subscription::Subscription - 
+*/
+Subscription::Subscription
+(
+  const std::string&        _tenant,
+  const std::string&        _servicePath,
+  SubscribeContextRequest*  scrP,
+  std::string               _subscriptionId,
+  int64_t                   _expiration,
+  Format                    _format
+)
+{
+  unsigned int ix;
+
+  LM_M(("KZ: Creating Subscription from SubscribeContextRequest"));
```

Comment:
Traces to be removed within this PR? Or they will stay, as in the create-entitie PR?